### ISSUE TITLE
macOS crash with 0.5.0

### ISSUE THREAD

**Issue by mliberty1** [jetperch/pyjoulescope#5]
When running the UI application, observed a crash when physically unplugging the Joulescope:

```
Thread 8 Crashed:
0   libusb-1.0.dylib              	0x00000001155a1f99 darwin_handle_transfer_completion + 436
1   libusb-1.0.dylib              	0x000000011559f3d9 handle_events + 1012
2   libusb-1.0.dylib              	0x000000011559ed7b libusb_handle_events_timeout_completed + 262
3   libusb-1.0.dylib              	0x000000011559b8d5 libusb_exit + 334
4   _ctypes.cpython-37m-darwin.so 	0x0000000108fc536f ffi_call_unix64 + 79
5   _ctypes.cpython-37m-darwin.so 	0x0000000108fc5b35 ffi_call + 713
6   _ctypes.cpython-37m-darwin.so 	0x0000000108fc13ea _ctypes_callproc + 647
7   _ctypes.cpython-37m-darwin.so 	0x0000000108fbbe1b PyCFuncPtr_call + 500
8   Python                        	0x0000000108ad3918 _PyObject_FastCallKeywords + 358
9   Python                        	0x0000000108b68ef0 call_function + 746
```

Only occurred once in about 80 unplug/plug cycles which is greatly improved over 0.4.6.

**Comment by mliberty1** [comment]
I found a setup that significantly increased this failure rate to around 50%.  Setup:
* macOS Mojave 10.14.5 on 13" MacBook Pro 2018
* nonda MI22SGRN [[Amazon](https://www.amazon.com/gp/product/B015Z7XE0A)] USB-C to USB-A adapter
* Anker A7516 4-Port Ultra Slip USB 3.0 Data Hub

**Comment by mliberty1** [comment]
Fixed in d65b7ef.  Testing with the "bad" setup no longer causes segmentation faults.  Also added faulthandler to the UI logging to assist with future segmentation fault troubleshooting.
